components/RevisionTimeline/MiniRevisionTimeline.tsx

```diff
@@ -1,14 +1,34 @@
+import axios from 'axios';
+import {
+  InProgressSubmission,
+  Submission,
+} from 'data/flexibleForms/forms.types';
 import { format } from 'date-fns';
-import { Submission } from 'data/flexibleForms/forms.types';
+import { useEffect, useState } from 'react';
 import s from './MiniRevisionTimeline.module.scss';
 
 interface Props {
-  submission: Submission;
+  inProgressSubmission: InProgressSubmission;
 }
 
-const RevisionTimeline = ({ submission }: Props): React.ReactElement | null => {
+const RevisionTimeline = ({
+  inProgressSubmission,
+}: Props): React.ReactElement | null => {
+  const [submission, setSubmission] = useState<Submission>();
+
+  useEffect(() => {
```

Comment:
We usually use SWR for fetching data in components, instead of calling axios directly from here. I think there's some good reasons for doing that too. One of which is that it has built in pagination logic etc., but there's definitely other pros.

Any reason why we're not doing that here?